### PR TITLE
Fix Tailwind configuration and use Turbopack

### PR DESCRIPTION
**Pull request by leerob**
# Update Next.js Configuration and Dependencies

## Overview
This pull request introduces several updates to the Next.js application configuration, dependency versions, and TypeScript settings. These changes aim to enhance the application's structure, compatibility with newer libraries, and overall development experience.

## Changes
- **Refactoring:**
    - Removed the PWA configuration from `next.config.mjs`, simplifying the configuration.
    - Updated the Tailwind CSS configuration in `tailwind.config.ts` to include specific directories for component scanning, improving clarity and organization.
- **Other Changes:**
    - Updated the TypeScript reference URL in `next-env.d.ts` to point to the latest documentation.
    - Upgraded the Next.js version to `15.0.0-canary.112` and React version to `19.0.0-rc-68dbd84b-20240812` in `package.json`.
    - Modified the `dev` script in `package.json` to utilize the `--turbo` flag for improved performance during development.
- **New Features:**
    - While not explicitly mentioned, the upgrade to React 19 may introduce new features and performance improvements.

> ✨ Generated with love by [Kaizen](https://cloudcode.ai) ❤️

<details>
<summary>Original Description</summary>

</details>

### CODE REVIEW

File: apps/web/next.config.mjs [hunk 1]
```diff
@@ -9,17 +9,10 @@ const baseNextConfig = {
 		TELEGRAM_BOT_TOKEN: process.env.TELEGRAM_BOT_TOKEN,
 	},
 	eslint: {
-		ignoreDuringBuilds: true
+		ignoreDuringBuilds: true,
 	},
 };
 
-const withPWA = pwa({
```
Comment (MaheshtheDev) [inline apps/web/next.config.mjs:16]: @leerob , may i know why PWA configuration has been removed?

Comment (leerob) [reply apps/web/next.config.mjs:16]: - next-pwa uses Workbox
- Workbox is no longer maintained
- next-pwa requires custom webpack configuration, making it incompatible with turbopack

I wanted to test on this PR if Turbopack works for you all, and what the performance improvements are. You might consider exploring alternative options based on what you were hoping to get from a PWA (was it install to your home screen? E.g. this is now built-into iOS).

Comment (MaheshtheDev) [reply apps/web/next.config.mjs:16]: next-pwa was integrated with thought of adding push notifications in near future

File: apps/web/package.json [hunk 2]
```diff
@@ -27,13 +27,12 @@
...
-		"next": "^14.1.1",
-		"next-pwa": "^5.6.0",
+		"next": "15.0.0-canary.112",
 		"novel": "^0.4.2",
 		"nuqs": "^1.17.4",
 		"posthog-js": "^1.154.5",
-		"react": "^18.2.0",
-		"react-dom": "^18.2.0",
+		"react": "19.0.0-rc-68dbd84b-20240812",
+		"react-dom": "19.0.0-rc-68dbd84b-20240812",
```
[the remark this answers is not in the document]
Comment (Dhravya) [reply apps/web/package.json:35]: !feedback removed important deps

File: apps/web/next.config.mjs (hunk 1 above)
Comment (Dhravya) [reply apps/web/next.config.mjs:16]: what's a good alternative to next-pwa?

**Comment by Dhravya** [comment]
Just tried to run it locally, not sure why the imports don't resolve anymore 😅
image

```
 ✓ Compiled / in 31.3s
 ⨯ ./apps/web/app/(landing)/CardPatterns/AnimatedBeamWithOutput.tsx.tsx
Module not found: Can't resolve './AnimatedBeamWithOutput.tsx.tsx'


https://nextjs.org/docs/messages/module-not-found



./apps/web/app/(landing)/Features.tsx.tsx
Module not found: Can't resolve './Features.tsx.tsx'


https://nextjs.org/docs/messages/module-not-found



./apps/web/app/(landing)/Hero.tsx.tsx
Module not found: Can't resolve './Hero.tsx.tsx'


https://nextjs.org/docs/messages/module-not-found



./apps/web/app/(landing)/Navbar.tsx.tsx
Module not found: Can't resolve './Navbar.tsx.tsx'


https://nextjs.org/docs/messages/module-not-found



./apps/web/app/(landing)/RotatingIcons.tsx.tsx
Module not found: Can't resolve './RotatingIcons.tsx.tsx'


https://nextjs.org/docs/messages/module-not-found



./apps/web/app/(landing)/Showcase.tsx.tsx
Module not found: Can't resolve './Showcase.tsx.tsx'


https://nextjs.org/docs/messages/module-not-found



./apps/web/app/global-error.tsx.tsx
Module not found: Can't resolve './global-error.tsx.tsx'


https://nextjs.org/docs/messages/module-not-found



./apps/web/app/providers.tsx.tsx
Module not found: Can't resolve './providers.tsx.tsx'


https://nextjs.org/docs/messages/module-not-found



./packages/ui/shadcn/toaster.tsx.tsx
Module not found: Can't resolve './toaster.tsx.tsx'


https://nextjs.org/docs/messages/module-not-found



./packages/ui/lib/utils.ts:2:1
Module not found: Can't resolve 'tailwind-merge'
  1 | import { type ClassValue, clsx } from "clsx";
> 2 | import { twMerge } from "tailwind-merge";
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  3 |
  4 | export function cn(...inputs: ClassValue[]) {
  5 |   return twMerge(clsx(inputs));



https://nextjs.org/docs/messages/module-not-found


 ○ Compiling /_error ...
 ✓ Compiled /_error in 6.1s
 GET / 500 in 38540ms
 ```

**Comment by RamGoel** [comment]
@Dhravya what's this .tsx.tsx in imports ? (Just curious)

**Comment by Dhravya** [comment]
@RamGoel I'm not sure either, maybe it's resolving the dependencies differently, out right now but will look into it

File: apps/web/next.config.mjs (hunk 1 above)
Comment (Tejs1) [reply apps/web/next.config.mjs:16]: This will help https://serwist.pages.dev/docs/next

**Comment by CodeTorso** [comment]
would love to see the performace benefits but 
image

**Comment by BlankParticle** [comment]
> @RamGoel I'm not sure either, maybe it's resolving the dependencies differently, out right now but will look into it

I ran into this issue while trying to implement million lint into our codebase, it doesn't work with turbopack
If you want million lint, turbopack is not an option for you

**Comment by Dhravya** [comment]
update: @aidenybai is helping fix this error

**Pull request merged**